spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -437,6 +433,24 @@ protected void actualStart() throws Exception {
 		}
 	}
 
+	protected void doRedeclareElementsIfNecessary() {
+		String routingLookupKey = getRoutingLookupKey();
+		if (routingLookupKey != null) {
+			SimpleResourceHolder.bind(getRoutingConnectionFactory(), routingLookupKey);
```

Comment:
True, but I think any meaningful implementation must use a thread local - there is no other way to pass context into the create connection request.

I am not too concerned about the overhead here - it's only done during initialization/restart. The concern about the SI poller last week was because it was applied to every poll.